filters/ratelimit/ratelimit.go

```diff
@@ -23,8 +25,10 @@ type spec struct {
 }
 
 type filter struct {
-	settings ratelimit.Settings
-	provider RatelimitProvider
+	mu         sync.Mutex
+	settings   ratelimit.Settings
+	provider   RatelimitProvider
+	overwrites map[string]ratelimit.Settings
```

Comment:
This would be reset on route table update, wouldn't it?

Reply:
It is also instance local

Reply:
The POC assumes that the backend is constantly telling in the response path the "current" changed settings, so yes and yes, but it's just a POC. The question would be if this is a reasonable assumption that the backend responses with the decision like this or if the backend just response once with a change.